lib/jekyll/commands/doctor.rb

```diff
@@ -91,6 +94,15 @@ def urls_only_differ_by_case(site)
           urls_only_differ_by_case
         end
 
+        def proper_site_url?(site)
+          url = site.config["url"]
+          [
+            url_exists?(url),
+            url_valid?(url),
```

Comment:
I get you perfectly and as it's impossible to tell if a blank `site.url` is intentional or not, you prefer to warn in both cases.

Maybe just make the messages more clear then? 

```sh
bundle exec jekyll doctor
Warning: You didn't set an URL in the config file, you may encounter problems with some plugins.
```
```sh
bundle exec jekyll doctor
Warning: The site URL does not seem to be valid, check the value of `url` in your config file.
```

```sh
bundle exec jekyll doctor
Warning: Your site URL does not seem to be absolute, check the value of `url` in your config file.
```